openmrs/src/main/java/org/motechproject/openmrs/resource/impl/BaseResource.java

```diff
@@ -85,6 +88,9 @@ protected void delete(Config config, String path, Object... params) {
 
     private ResponseEntity<String> exchange(Config config, URI url, HttpMethod method, String body, HttpHeaders headers) {
         headers.add("Authorization", "Basic " + prepareCredentials(config));
+
+        httpClient.getState().clearCookies();
```

Comment:
HttpClient class has deprecated method to set cookies policy, but it doesn't cover our requirements. I could use newer way to setup HttpClient, but this require another dependency.